### ISSUE TITLE
AttributeError: module 'tornado.ioloop' has no attribute '_Selectable'

### ISSUE THREAD

**Issue by nickcorona** [sfu-db/dataprep#218]
AttributeError: module 'tornado.ioloop' has no attribute '_Selectable'

**Comment by jinglinpeng** [comment]
Hi @nickcorona , could you please share the code that causes the error?

**Comment by nickcorona** [comment]
`from dataprep import eda` causes the error

**Comment by brandonlockhart** [comment]
This is arising in Kaggle notebooks. With
``` python
!pip install dataprep
from dataprep import eda
```
this is the stack trace
``` python
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-4-4816e3b307f4> in <module>
----> 1 from dataprep import eda

/opt/conda/lib/python3.7/site-packages/dataprep/eda/__init__.py in <module>
      6 
      7 from bokeh.io import output_file, output_notebook
----> 8 from .basic import compute, plot, render
      9 from .correlation import compute_correlation, plot_correlation, render_correlation
     10 from .missing import compute_missing, plot_missing, render_missing

/opt/conda/lib/python3.7/site-packages/dataprep/eda/basic/__init__.py in <module>
     10 
     11 from .compute import compute
---> 12 from .render import render
     13 from ..report import Report
     14 from ..dtypes import DTypeDef

/opt/conda/lib/python3.7/site-packages/dataprep/eda/basic/render.py in <module>
     39 from ..dtypes import Continuous, DateTime, Nominal, is_dtype
     40 from ..intermediate import Intermediate
---> 41 from ..palette import BIPALETTE, PALETTE
     42 
     43 __all__ = ["render"]

/opt/conda/lib/python3.7/site-packages/dataprep/eda/palette.py in <module>
      4 # pylint: disable=no-name-in-module
      5 from bokeh.palettes import Category20  # type: ignore
----> 6 from holoviews.plotting.util import process_cmap
      7 
      8 PALETTE = Category20[20]

/opt/conda/lib/python3.7/site-packages/holoviews/__init__.py in <module>
     10 
     11 from . import util                                       # noqa (API import)
---> 12 from .annotators import annotate                         # noqa (API import)
     13 from .core import archive, config                        # noqa (API import)
     14 from .core.boundingregion import BoundingBox             # noqa (API import)

/opt/conda/lib/python3.7/site-packages/holoviews/annotators.py in <module>
      8 import param
      9 
---> 10 from panel.pane import PaneBase
     11 from panel.layout import Row, Tabs
     12 from panel.util import param_name

/opt/conda/lib/python3.7/site-packages/panel/__init__.py in <module>
      3 import param as _param
      4 
----> 5 from . import layout # noqa
      6 from . import links # noqa
      7 from . import pane # noqa

/opt/conda/lib/python3.7/site-packages/panel/layout.py in <module>
     22 from .io.state import state
     23 from .util import param_name, param_reprs
---> 24 from .viewable import Layoutable, Reactive
     25 
     26 _row = namedtuple("row", ["children"])

/opt/conda/lib/python3.7/site-packages/panel/viewable.py in <module>
     25 
     26 from .callbacks import PeriodicCallback
---> 27 from .config import config, panel_extension
     28 from .io.embed import embed_state
     29 from .io.model import add_to_doc, hold, patch_cds_msg

/opt/conda/lib/python3.7/site-packages/panel/config.py in <module>
     18                          extension as _pyviz_extension)
     19 
---> 20 from .io.notebook import load_notebook
     21 from .io.state import state
     22 

/opt/conda/lib/python3.7/site-packages/panel/io/__init__.py in <module>
      8 from .model import add_to_doc, remove_root, diff # noqa
      9 from .resources import Resources # noqa
---> 10 from .server import get_server, serve, unlocked # noqa
     11 from .notebook import block_comm, ipywidget, load_notebook, push # noqa

/opt/conda/lib/python3.7/site-packages/panel/io/server.py in <module>
     15 
     16 from bokeh.document.events import ModelChangedEvent
---> 17 from bokeh.server.server import Server
     18 from tornado.websocket import WebSocketHandler
     19 from tornado.web import RequestHandler

/opt/conda/lib/python3.7/site-packages/bokeh/server/server.py in <module>
     39 # External imports
     40 from tornado import version as tornado_version
---> 41 from tornado.httpserver import HTTPServer
     42 from tornado.ioloop import IOLoop
     43 

/opt/conda/lib/python3.7/site-packages/tornado/httpserver.py in <module>
     30 
     31 from tornado.escape import native_str
---> 32 from tornado.http1connection import HTTP1ServerConnection, HTTP1ConnectionParameters
     33 from tornado import httputil
     34 from tornado import iostream

/opt/conda/lib/python3.7/site-packages/tornado/http1connection.py in <module>
     32 from tornado import gen
     33 from tornado import httputil
---> 34 from tornado import iostream
     35 from tornado.log import gen_log, app_log
     36 from tornado.util import GzipDecompressor

/opt/conda/lib/python3.7/site-packages/tornado/iostream.py in <module>
    227 
    228 
--> 229 class BaseIOStream(object):
    230     """A utility class to write to and read from a non-blocking file or socket.
    231 

/opt/conda/lib/python3.7/site-packages/tornado/iostream.py in BaseIOStream()
    303         self._closed = False
    304 
--> 305     def fileno(self) -> Union[int, ioloop._Selectable]:
    306         """Returns the file descriptor for this stream."""
    307         raise NotImplementedError()

AttributeError: module 'tornado.ioloop' has no attribute '_Selectable'
```

**Comment by nickcorona** [comment]
Any update on this bug?

**Comment by jnwang** [comment]
Sorry for the late reply. @jinglinpeng is working on it. We will do our best to get it fixed in 0.2.9. If not, we will give you an update on our effort.

Thanks for using DataPrep!

**Comment by jinglinpeng** [comment]
Hi @nickcorona , the issue is caused by the boketh version in Kaggle. @dovahcrow is working on this. Will give you an update once it is fixed.

**Comment by dovahcrow** [comment]
Hi @nickcorona the fix will be landed in the next release 0.2.10.

**Comment by nickcorona** [comment]
> Hi @nickcorona the fix will be landed in the next release 0.2.10.

Great thanks!
